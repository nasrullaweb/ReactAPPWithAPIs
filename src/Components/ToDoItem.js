import React, { Component } from 'react'

class ToDoItem extends Component {
    render(){
        return (
            <div className="todoListMain">
                <div className="header">
                    <form 
                        onSubmit={(event) => this.props.addItem(event)} 
                        onReset={(event)=>this.props.cancelAction(event)} >
                        <input type="text" className="taskInputVal" placeholder="Add to do item" 
                            onChange={(event) => this.props.handleInputChange(event)} 
                            value={this.props.inputText.text} 
                        />
                        <button>{this.props.upadetText ? 'Update': 'Add'}</button>
                        <button type="reset" className={this.props.upadetText ? 'displayblock' : 'displaynone'} >Cancel</button>
                    </form>
                </div>
            </div>
        )
    }
}

export default ToDoItem;