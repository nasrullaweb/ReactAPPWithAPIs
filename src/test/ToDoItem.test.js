import React from 'react';
import ToDoItem from '.././Components/ToDoItem';
import renderer from 'react-test-renderer';

test('Render todo form', () => {
    const component = renderer.create(
        <div className="todoListMain">
        <div className="header">
            <form>
                <input type="text" value="the value"/>
                <button>Add</button>
                <button>Cancel</button>
            </form>
        </div>
    </div>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });