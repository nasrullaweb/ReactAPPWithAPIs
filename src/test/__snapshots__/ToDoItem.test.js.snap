// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render todo form 1`] = `
<div
  className="todoListMain"
>
  <div
    className="header"
  >
    <form>
      <input
        type="text"
        value="the value"
      />
      <button>
        Add
      </button>
      <button>
        Cancel
      </button>
    </form>
  </div>
</div>
`;
